internal/contour/handler.go

```diff
@@ -189,19 +183,12 @@ func (e *EventHandler) onUpdate(op interface{}) bool {
 	case opAdd:
 		return e.builder.Source.Insert(op.obj)
 	case opUpdate:
-		if cmp.Equal(op.oldObj, op.newObj,
-			cmpopts.IgnoreFields(contour_api_v1.HTTPProxy{}, "Status"),
-			cmpopts.IgnoreFields(contour_api_v1alpha1.ExtensionService{}, "Status"),
-			cmpopts.IgnoreFields(gatewayapi_v1beta1.GatewayClass{}, "Status"),
-			cmpopts.IgnoreFields(gatewayapi_v1beta1.Gateway{}, "Status"),
-			cmpopts.IgnoreFields(gatewayapi_v1beta1.HTTPRoute{}, "Status"),
-			cmpopts.IgnoreFields(gatewayapi_v1alpha2.TLSRoute{}, "Status"),
-			cmpopts.IgnoreFields(metav1.ObjectMeta{}, "ResourceVersion"),
-			cmpopts.IgnoreFields(metav1.ObjectMeta{}, "ManagedFields"),
-		) {
+		// Ignore if generation is the same.
+		if op.oldObj.(client.Object).GetGeneration() == op.newObj.(client.Object).GetGeneration() {
```

Comment:
I'm observing that in the old code, we did not ignore status for things like Services, Secrets, etc. Wondering if there's any particular need to trigger DAG rebuilds when status for those referenced resources changes? The only thing I can think of off the top of my head is the Envoy service's status.loadBalancer changing which should trigger an update to Ingress/HTTPProxy status, but that goes through a different code path than this.

Reply:
True, it was unfortunately not simple as this. Not all objects have generation field set at all.